src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/EnvelopeProcessorTest.java

```diff
@@ -1,132 +1,50 @@
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.ArgumentCaptor;
 import org.mockito.Mock;
-import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.junit.jupiter.MockitoExtension;
 import uk.gov.hmcts.reform.bulkscan.orchestrator.services.bulkscanprocessorclient.client.BulkScanProcessorClient;
-import uk.gov.hmcts.reform.bulkscan.orchestrator.services.bulkscanprocessorclient.exceptions.ReadEnvelopeException;
-import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.ReceiverProvider;
 
-import java.util.UUID;
+import java.util.concurrent.ExecutionException;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.BDDMockito.given;
-import static org.mockito.Mockito.never;
-import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 
-@RunWith(MockitoJUnitRunner.class)
-public class EnvelopeProcessorTest {
+@ExtendWith(MockitoExtension.class)
+class EnvelopeProcessorTest {
 
-    @Mock private BulkScanProcessorClient bulkScanProcessorClient;
-    @Mock private ReceiverProvider receiverProvider;
+    @Mock
+    BulkScanProcessorClient bulkScanProcessorClient;
 
-    @Mock private IMessage someMessage;
-    @Mock private IMessageReceiver receiver;
+    private EnvelopeProcessor envelopeProcessor;
 
-    private EnvelopeProcessor processor;
+    @Mock
+    private IMessage someMessage;
+    private static final String MSG_ID = "hello!";
 
-    @Before
-    public void setUp() throws Exception {
-        this.processor = new EnvelopeProcessor(receiverProvider, bulkScanProcessorClient);
-        given(receiverProvider.get()).willReturn(receiver);
+    @BeforeEach
+    void before() {
+        given(someMessage.getMessageId()).willReturn(MSG_ID);
+        envelopeProcessor = new EnvelopeProcessor(bulkScanProcessorClient);
     }
 
     @Test
-    public void should_read_all_messages_from_the_queue() throws Exception {
-        // given
-        // there are 2 messages on the queue
-        given(receiver.receive())
-            .willReturn(someMessage)
-            .willReturn(someMessage)
-            .willReturn(null);
-
-        // when
-        processor.run();
-
-        // then
-        verify(bulkScanProcessorClient, times(2)).getEnvelopeById(any());
-        verify(receiver, times(3)).receive();
-    }
-
-    @Test
-    public void should_use_queue_message_id_to_read_envelope() throws Exception {
-        // given
-        final String msgId = "hello!";
-        given(someMessage.getMessageId())
-            .willReturn(msgId);
-
-        given(receiver.receive())
-            .willReturn(someMessage)
-            .willReturn(null);
-
+    @DisplayName("should use queue message id to read envelope")
+    void readEnvelope() throws ExecutionException, InterruptedException {
```

Comment:
I personally think the intention of the test is more readable and it is the way forward for unit testing in java.
Otherwise I can just re-write it in kotlin with a better naming such as 
```kotlin
fun `this is the other way we can do this`(){
}
``` 
if you would prefer this ?

Reply:
Something for the retro. I vote for a change since I was never in favour of this strict naming convention rule on test cases. Nevertheless, as mentioned in review conclusion - let's not introduce junit5 for now - leave for entirely separate pr